### PR TITLE
Add a user menu

### PR DESCRIPTION
**Pull request by kasbah**

### CODE REVIEW

**Comment by AbdulrhmnGhanem** [comment]
## closed
image
# open
image
# small screen
image

**Comment by AbdulrhmnGhanem** [comment]
I guess the implementation for the `/settings` page will be in a different PR, right?

File: .env.example [hunk 1]
```diff
@@ -17,10 +17,6 @@ CYPRESS_GITEA_ADMIN_USERNAME=__admin
 CYPRESS_GITEA_ADMIN_PASSWORD=123456
 # Abort the rest of Cypress tests on first failure
 FAIL_FAST_ENABLED=true
-# Incremental static regeneration period(in seconds) for `/`
-INDEX_ISR_INTERVAL=3600
-# Incremental static regeneration period(in seconds) for `/projects/[user]`
-USER_ISR_INTERVAL=86400
```
Comment (kasbah) [inline .env.example:23]: Why are these being removed? Are they not being used anymore?

**Comment by AbdulrhmnGhanem** [comment]
> For that I think we need a clearer separation between the user menu and the rest of the options.

like this?
image

**Comment by kasbah** [comment]
Yes, but that line is pretty ugly. I think there is a separator in Semantic UI that you can use. Also, I think the user stuff should be up top in the user menu.

File: frontend/src/components/NavBar/UserMenu.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import Link from 'next/link'
+import Image from 'next/image'
+import { useRouter } from 'next/router'
+
+import styles from './UserMenu.module.scss'
+import { AuthContext } from '@contexts/AuthContext'
+
+export const UserDropDownMenu = () => {
+  const { user } = useContext(AuthContext)
+  const [open, setOpen] = useState(false)
```
Comment (kasbah) [inline frontend/src/components/NavBar/UserMenu.js:12]:
```suggestion
  const [isOpen, setOpen] = useState(false)
```

**Comment by AbdulrhmnGhanem** [comment]
I increased the height. Though increasing the width to make as the pink rectangle will distort the image

image

**Pull request merged**